File: go/builder/HttpRequestBuilder.go
package main

import (
	"errors"
	"time"
)

type HttpRequestBuilder struct {
	config HttpRequestConfig
}

func NewHttpRequestBuilder(url string) *HttpRequestBuilder {
	return &HttpRequestBuilder{
		config: HttpRequestConfig{
			URL:     url,
			Method:  GET,
			Headers: make(map[string]string),
			Timeout: 5 * time.Second,
			Retries: 0,
		},
	}
}

func (b *HttpRequestBuilder) SetURL(url string) *HttpRequestBuilder {
	b.config.URL = url
	return b
}

func (b *HttpRequestBuilder) SetMethod(method HttpMethod) *HttpRequestBuilder {
	b.config.Method = method
	return b
}

func (b *HttpRequestBuilder) SetHeader(key, value string) *HttpRequestBuilder {
	b.config.Headers[key] = value
	return b
}

func (b *HttpRequestBuilder) SetBody(body any) *HttpRequestBuilder {
	b.config.Body = body
	return b
}

func (b *HttpRequestBuilder) SetTimeout(timeout time.Duration) *HttpRequestBuilder {
	b.config.Timeout = timeout
	return b
}

func (b *HttpRequestBuilder) SetRetries(retries int) *HttpRequestBuilder {
	b.config.Retries = retries
	return b
}

func (b *HttpRequestBuilder) Build() (*HttpRequest, error) {
	if b.config.URL == "" {
		return nil, errors.New("URL is required")
	}
	return &HttpRequest{config: b.config}, nil
}

File: python/builder/HttpRequestBuilder.py
from HttpRequestConfig import HttpMethod, HttpRequestConfig
from HttpRequest import HttpRequest
from typing import Any

class HttpRequestBuilder:
    def __init__(self, url: str):
        self._config = HttpRequestConfig(url=url)

    def set_url(self, url: str) -> "HttpRequestBuilder":
        self._config.url = url
        return self

    def set_method(self, method: HttpMethod) -> "HttpRequestBuilder":
        self._config.method = method
        return self

    def set_header(self, key: str, value: str) -> "HttpRequestBuilder":
        self._config.headers[key] = value
        return self

    def set_body(self, body: Any) -> "HttpRequestBuilder":
        self._config.body = body
        return self

    def set_timeout(self, seconds: int) -> "HttpRequestBuilder":
        self._config.timeout = seconds
        return self

    def set_retries(self, count: int) -> "HttpRequestBuilder":
        self._config.retries = count
        return self

    def build(self) -> HttpRequest:
        if not self._config.url:
            raise ValueError("URL is required")

        return HttpRequest(self._config)
